internal/config/config.go

```diff
@@ -243,16 +243,20 @@ func For(resources ClusterResources, validate Validate) (*Config, error) {
 		if err != nil {
 			return nil, err
 		}
+		ipPools, err := selectPools(resources.Pools, l2Adv.Spec.IPAddressPoolSelectors)
+		if err != nil {
+			return nil, err
+		}
 		// No pool selector means select all pools
-		if len(l2Adv.Spec.IPAddressPools) == 0 {
+		if len(l2Adv.Spec.IPAddressPools) == 0 && len(ipPools) == 0 {
```

Comment:
This must be empty label selector.
If the user puts a non matching selector, we don't want to associate the advertisement to all pools

Reply:
valid point, changed it.